### PR TITLE
Remove deprecated legacy pulse builder commands

### LINKED ISSUE #11152: Deprecate legacy pulse builder command.
### What should we add?

Pulse builder still supports following commands that inject circuit gate operation into the pulse context: 
- `qiskit.pulse.builder.call_gate`
- `qiskit.pulse.builder.cx`
- `qiskit.pulse.builder.u1`
- `qiskit.pulse.builder.u2`
- `qiskit.pulse.builder.u3`
- `qiskit.pulse.builder.x`

These are legacy workflow which we implemented before the pulse gate feature. These command introduce weird cyclic dependency between circuit and pulse program, and must be deprecated before the next release. 

In addition, following commands are nice to be deprecated as well.

- `qiskit.pulse.builder.active_transpiler_settings`
- `qiskit.pulse.builder.active_circuit_scheduler_settings`
- `qiskit.pulse.builder.transpiler_settings`

### Background
Note that the standard program representation in Qiskit is now `QuantumCircuit`, and pulse programs exist as a microcode (calibration) of the circuit. The pulse builder is a tool to define such microprograms, and circuit -> pulse conversion must be explicitly done with the circuit scheduler. Injecting circuit instruction in pulse program is indeed an opposite workflow.

### PR DESCRIPTION
**Pull request by MozammilQ**
### Summary
This PR removes deprecated logic for legacy pulse builder commands.
Deprecation is done in this PR #11249 

This PR 

Removes
- `qiskit.pulse.builder.call_gate`
- `qiskit.pulse.builder.cx`
- `qiskit.pulse.builder.u1`
- `qiskit.pulse.builder.u2`
- `qiskit.pulse.builder.u3`
- `qiskit.pulse.builder.x`

- `qiskit.pulse.builder.active_transpiler_settings`
- `qiskit.pulse.builder.active_circuit_scheduler_settings`
-  `qiskit.pulse.builder.transpiler_settings`
- arguments `default_transpiler_settings` ,
                and `default_circuit_scheduler_settings` of 
               `qiskit.pulse.builder.build` function.
- `QuantumCircuit` type of the `target` argument in `qiskit.pulse.builder.call` function

Modifies:
- Module doc of `qiskit.pulse.builder.py` (example code with circuit elements)
- The example code 3 of function doc for `qiskit.pulse.builder.call`

and, modifies related tests in
- test/python/pulse/test_builder.py
- test/python/pulse/test_block.py
- test/python/pulse/test_builder_v2.py
- test/python/transpiler/test_calibrationbuilder.py

### Details and comments
Refer to Issue: https://github.com/Qiskit/qiskit/issues/11152

### CODE REVIEW

File: releasenotes/notes/Deprecate_legacy_pulse_builder_command-47651b395940e687.yaml
Comment (Raghav-Bell) [inline releasenotes/notes/Deprecate_legacy_pulse_builder_command-47651b395940e687.yaml]: Doctests are failing due to `sphinx` error. Following may help you to debug [Code block with line numbers problem](https://groups.google.com/g/sphinx-users/c/3VCVS73nhsg), [What is the point of the Sphinx highlight_language config option if code-block:: doesn't have an optional argument?](https://stackoverflow.com/questions/34739635/what-is-the-point-of-the-sphinx-highlight-language-config-option-if-code-block)

Comment (MozammilQ) [reply releasenotes/notes/Deprecate_legacy_pulse_builder_command-47651b395940e687.yaml]: Hi! Thanks, @Raghav-Bell 
could you please explain to me how did you find the error?
Please, help me know what tools do i use to find out errors like these?

Comment (Raghav-Bell) [reply releasenotes/notes/Deprecate_legacy_pulse_builder_command-47651b395940e687.yaml]: @MozammilQ, Check failed CI logfiles for debugging PR.

Comment (MozammilQ) [reply releasenotes/notes/Deprecate_legacy_pulse_builder_command-47651b395940e687.yaml]: @Raghav-Bell , I always check the log files, but the failed log file (for `Run Docs build`) showed this:

Screenshot_20231104_154409

How did you pin point the error at `.. code-block:: python`
please, help! :) thanks for the reply :)

Comment (Raghav-Bell) [reply releasenotes/notes/Deprecate_legacy_pulse_builder_command-47651b395940e687.yaml]: I search for keywords like error or failed. Somewhere in the logfile you may get error printed.

Comment (Raghav-Bell) [reply releasenotes/notes/Deprecate_legacy_pulse_builder_command-47651b395940e687.yaml]: image

File: qiskit/pulse/builder.py
Comment (nkanazawa1989) [inline qiskit/pulse/builder.py]:
```suggestion
@deprecate_func(since="0.46.0")
```
@mtreinish Is there any guide for deprecation? Do we need to open two PRs? One to https://github.com/Qiskit/qiskit/tree/stable/0.46 with deprecation warning and another one to main branch with code removal?

**Review by nkanazawa1989 — COMMENTED** [review]
Thanks @MozammilQ ! Could you please deprecate/update following places too?

- Module doc of `qiskit.pulse.builder.py` (example code with circuit elements)
- Several args (default_transpiler_settings, default_circuit_scheduler_settings) in `qiskit.pulse.builder.build` function
- `QuantumCircuit` type of the `target` argument in `qiskit.pulse.builder.call` function (you can use predicate feature of `deprecate_arg` decorator). The example code 3 of function doc too.

File: qiskit/pulse/builder.py
Comment (ElePT) [reply qiskit/pulse/builder.py]: Yes, you will need 2 PRs, one with the deprecation (to 0.46) and one with the removal (to 1.0 == `main`), and the deprecation warnings should say "since 0.46" (as suggested by @nkanazawa1989). In case you are considering it, I think that there's no need to close and re-open it against `stable/0.46`, we can merge it to main and backport it to 0.46. (now that I think about it, this might actually be the recommended course of action, because else we would have diverging histories, @mtreinish?). It would be a good idea to write this down and have a sort of "guide" we can refer people to, because I expect it to be confusing for many contributors.

Comment (mtreinish) [reply qiskit/pulse/builder.py]: The pattern we've been following is just opening a PR on `stable/0.46` for the deprecation paired with a removal PR on `main` for 1.0.0. This was mostly done to minimize CI time, it's only 2 CI runs instead of 3 to deprecate and remove. They can also be pushed in parallel without any dependency between PRs.

But, I wouldn't say we're committed to that as a general requirement, doing both a deprecation to be backported to `stable/0.46` followed by an immediate removal PR to main would also work fine. The only complication is for backport you'd need to use a comment to manually tell mergify to backport to `stable/0.46`. For example, leaving a comment like: `@Mergifyio backport stable/0.46` after the PR merges. Using the stable backport potential tag in this case will trigger a backport to stable/0.45 which is incorrect for new deprecations.

In the interest of documenting this I pushed up https://github.com/Qiskit/qiskit/pull/11205 to document the branching versioning strategy and all the general deprecation procedures around this. But I didn't put in specifics around how to handle PRs for this situation. I think we can updating `CONTRIBUTING.md` after #11205 merges if we feel more guidance is needed on the specifics around this. In general backporting and stable branch policy will be getting more complicated moving forward as we support > 1 stable branch at a time (this is just the first example of it).

**Comment by MozammilQ** [comment]
@nkanazawa1989 , please see if this is ok :)

**Comment by nkanazawa1989** [comment]
Thanks @MozammilQ the PR itself looks good to me. Unfortunately (fortunately?) we are now on bit tricky period for the first major version release. This means `main` branch doesn't accept code deprecation, and they should go to `stable/0.46` branch instead. Guide is here: https://github.com/Qiskit/feedback/discussions/36

Since this PR targets the `main` branch, you can immediately remove all code to deprecate, and move the deprecation code you wrote for the PR to another PR targeting `stable/0.46`. If you feel this workflow is troublesome, I'll merge this PR as-is and take care of the rest of procedure. What do you prefer?

**Comment by MozammilQ** [comment]
@nkanazawa1989 , I will first move the deprecation code which is currently in this PR to another PR against `stable/0.46`,
and then, remove the deprecated code here in this PR.

**Comment by MozammilQ** [comment]
@nkanazawa1989 , PR #11249 , is against stable/0.46,
now, I will remove the deprecated code here, (against main)

**Review by nkanazawa1989 — COMMENTED** [review]
Thanks @MozammilQ I think you need to delete removed usage of builder command from the class and module docs, rather than adding warning message in this PR. I'll merge this PR once after docs errors are resolved.

**Comment by MozammilQ** [comment]
@nkanazawa1989 , please see if this is good enough :)

**Review by nkanazawa1989 — COMMENTED** [review]
Could you please also remove `self._circuit_scheduler_settings` and command? There is no circuit input to be scheduled.

**Comment by MozammilQ** [comment]
@nkanazawa1989 , please see if this is good enough :)

**Review by nkanazawa1989 — APPROVED** [review]
Thanks @MozammilQ all looks good! I'll merge the PR.

**Pull request merged**
